tests/console/validate_lvm_with_raid1.pm

```diff
@@ -0,0 +1,98 @@
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "y2_module_consoletest";
+use testapi;
+use utils;
+use Test::Assert ':all';
+use scheduler 'get_test_data';
+
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+
+    my $config = get_test_data();
+    $config->{expected_num_devs} = scalar keys %{$config->{disks}};
+    _check_lvm_partitioning($config);
+    _check_raid1_partitioning($config);
+    _remove_raid_disk($config);
+    _reboot();
+    $self->wait_boot;
+    _check_raid_disks_after_reboot($config);
+    _restore_raid_disk($config);
+}
+
+sub _check_lvm_partitioning {
+    my ($config) = @_;
+    my $activelvm = script_output q[lvscan | awk '{print $2}' | sed s/\'//g];
+    assert_equals $config->{lvm}->{lvpath}, $activelvm, "lv name is not the same. \n" . script_output q[lvscan];
+    my $pvname = script_output q[pvscan -s | grep '/dev'];
+    assert_equals $config->{lvm}->{pvname}, $pvname, "pv name is not the same.";
+}
+
+sub _check_raid1_partitioning {
+    my ($config) = @_;
+    record_info("raid1 name", $config->{raid1}->{name});
+    assert_script_run 'mdadm --detail ' . $config->{lvm}->{pvname};    # should correspond to $config->{raid1}->{name}
+
+    assert_script_run 'grep \'md0 : active ' . $config->{raid1}->{level} . '\' /proc/mdstat';
+
+    my $active_devs = script_output("mdadm --detail " . $config->{raid1}->{level} . " |grep \"Active Devices\" |awk '{ print \$4 }'");
+    assert_equals($active_devs, $config->{expected_num_devs}, "Active devices are different");
+
+    for (keys %{$config->{disks}}) {
+        my $line = script_output "mdadm --detail " . $config->{raid1}->{level} . " | awk '{ if((/$_/) && (\$5 ~ /active/) && (\$6 ~ /sync/)) {print}}'";
+        assert_not_null($line, "$_ not active");
+    }
+}
+
+
+sub _remove_raid_disk {
+    my ($config) = @_;
+    assert_script_run 'mdadm --manage ' . $config->{raid1}->{level} . ' --set-faulty ' . $config->{raid1}->{disks}[3];
+    $config->{expected_num_devs}--;
+    my $active_devs = script_output("mdadm --detail " . $config->{raid1}->{level} . " |grep \"Active Devices\" |awk '{ print \$4 }'");
+    assert_equals($config->{expected_num_devs}, $active_devs, "Active devices are different after set faulty device");
+
+    assert_script_run 'mdadm --manage ' . $config->{raid1}->{level} . ' --remove /dev/vdd2';
+    script_retry "mdadm --detail " . $config->{raid1}->{level} . " | grep 'Active Devices : 3'";
+    my $removedDisk = script_output "mdadm --detail " . $config->{raid1}->{level} . " | awk '{ if((\$4 == " . $config->{expected_num_devs} . ") && (\$5 ~ /removed/)) {print}}'";
+    assert_not_null($removedDisk, "$removedDisk should have been removed");
+}
+
+sub _reboot {
+    record_info('system reboots');
+    type_string "shutdown -r now\n";
+}
+
+sub _check_raid_disks_after_reboot {
+    my ($config) = @_;
+    select_console 'root-console';
+
+    # check again
```

Comment:
Same than below, now with better name of the function, no need for the comment I guess.

Reply:
removed